### PR TITLE
Laszlo.kardinal/form refactor

### PR DESCRIPTION
**Pull request by laszlokardinal-dina**

### CODE REVIEW

**Comment by ghatighorias** [comment]
Maybe from the next commit you might want to break them into smaller pieces, cause altogether 2150 lines are too much to review

**Comment by rhnonose** [comment]
Same feeling as @ghatighorias, specially if involves a lot of JS stuff that's harder.

File: web/templates/course/form.html.eex [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
+                <p>Oops, something went wrong! Please check the errors below.</p>
+              </div>
+            <% end %>
+            <%= CoursePlanner.SharedView.form_text f, :name, required: true %>
+            <%= CoursePlanner.SharedView.form_textarea f, :description, required: true %>
+          <% end %>
+          <%= CoursePlanner.SharedView.card_actions do %>
+            <div class="row row--vspace">
+              <div class="col-xs-12 col-md-3">
+                <%= CoursePlanner.SharedView.form_button "Back", student_path(@conn, :index) %>
```
Comment (ggpasqualino) [inline web/templates/course/form.html.eex]: this should go to `course_path(@conn, :index)`

File: web/templates/term/form.html.eex [hunk 2]
```diff
@@ -1,38 +1,68 @@
...
+              <div class="col-xs-12 col-md-3">
+                <%= link_to_holiday_fields(f, :holidays) %>
+              </div>
+            </div>
+
+          <% end %>
+          <%= CoursePlanner.SharedView.card_actions do %>
+            <div class="row row--vspace">
+              <div class="col-xs-12 col-md-3">
+                <%= CoursePlanner.SharedView.form_button "Back", student_path(@conn, :index) %>
```
Comment (ggpasqualino) [inline web/templates/term/form.html.eex]: this should go to `term_path(@conn, :index)`

File: web/templates/offered_course/form.html.eex [hunk 3]
```diff
@@ -1,48 +1,47 @@
...
+            %>
+            <%= CoursePlanner.SharedView.form_multiselect f, :teacher_ids, teachers, selected: selected_teachers(@changeset),
+                                                                                     label: "Teachers",
+                                                                                     tooltip_text: "Add new teacher"
+            %>
+          <% end %>
+          <%= CoursePlanner.SharedView.card_actions do %>
+            <div class="row row--vspace">
+              <div class="col-xs-12 col-md-3">
+                <%= CoursePlanner.SharedView.form_button "Back", student_path(@conn, :index) %>
```
Comment (ggpasqualino) [inline web/templates/offered_course/form.html.eex]: this should go to `offered_course_path(@conn, :index)`

File: web/templates/class/form.html.eex [hunk 4]
```diff
@@ -1,40 +1,35 @@
...
+            %>
+            <%= CoursePlanner.SharedView.form_date f, :date, required: true %>
+            <%= CoursePlanner.SharedView.form_time f, :starting_at, required: true %>
+            <%= CoursePlanner.SharedView.form_time f, :finishes_at, required: true %>
+            <%= CoursePlanner.SharedView.form_text f, :classroom %>
+          <% end %>
+          <%= CoursePlanner.SharedView.card_actions do %>
+            <div class="row row--vspace">
+              <div class="col-xs-12 col-md-3">
+                <%= CoursePlanner.SharedView.form_button "Back", student_path(@conn, :index) %>
```
Comment (ggpasqualino) [inline web/templates/class/form.html.eex]: this should go to `class_path(@conn, :index)`

File: web/templates/task/form.html.eex [hunk 5]
```diff
@@ -1,35 +1,32 @@
...
+            <% end %>
+            <%= CoursePlanner.SharedView.form_text f, :name, required: true %>
+            <%= CoursePlanner.SharedView.form_datetime f, :start_time, required: true %>
+            <%= CoursePlanner.SharedView.form_datetime f, :finish_time, required: true %>
+            <%= CoursePlanner.SharedView.form_select f, :user_id, users, label: "Assignee" %>
+          <% end %>
+          <%= CoursePlanner.SharedView.card_actions do %>
+            <div class="row row--vspace">
+              <div class="col-xs-12 col-md-3">
+                <%= CoursePlanner.SharedView.form_button "Back", student_path(@conn, :index) %>
```
Comment (ggpasqualino) [inline web/templates/task/form.html.eex]: this should go to `task_path(@conn, :index)`

File: web/templates/volunteer/form.html.eex [hunk 6]
```diff
@@ -1,41 +1,34 @@
...
+              <div class="form-error">
+                <p>Oops, something went wrong! Please check the errors below.</p>
+              </div>
+            <% end %>
+            <%= CoursePlanner.SharedView.form_text f, :name %>
+            <%= CoursePlanner.SharedView.form_text f, :family_name %>
+            <%= CoursePlanner.SharedView.form_text f, :nickname %>
+            <%= CoursePlanner.SharedView.form_text f, :email, required: true %>
+            <%= CoursePlanner.SharedView.form_text f, :phone_number %>
+            <%= CoursePlanner.SharedView.form_select f, :status, statuses %>
```
Comment (rhnonose) [inline web/templates/volunteer/form.html.eex]: We are removing status from the user, so this is going to break.

File: web/templates/term/form.html.eex [hunk 7]
```diff
@@ -1,38 +1,68 @@
...
+          <%= CoursePlanner.SharedView.card_content do %>
+            <%= if @changeset.action do %>
+              <div class="form-error">
+                <p>Oops, something went wrong! Please check the errors below.</p>
+              </div>
+            <% end %>
+            <%= CoursePlanner.SharedView.form_text f, :name, required: true %>
+            <%= CoursePlanner.SharedView.form_date f, :start_date, required: true %>
+            <%= CoursePlanner.SharedView.form_date f, :end_date, required: true %>
+            <%= CoursePlanner.SharedView.form_select f, :status, statuses, required: true %>
```
Comment (rhnonose) [inline web/templates/term/form.html.eex]: This status will be removed as well.

File: web/templates/teacher/form.html.eex [hunk 8]
```diff
@@ -1,41 +1,34 @@
...
+              <div class="form-error">
+                <p>Oops, something went wrong! Please check the errors below.</p>
+              </div>
+            <% end %>
+            <%= CoursePlanner.SharedView.form_text f, :name %>
+            <%= CoursePlanner.SharedView.form_text f, :family_name %>
+            <%= CoursePlanner.SharedView.form_text f, :nickname %>
+            <%= CoursePlanner.SharedView.form_text f, :email, required: true %>
+            <%= CoursePlanner.SharedView.form_text f, :phone_number %>
+            <%= CoursePlanner.SharedView.form_select f, :status, statuses %>
```
Comment (rhnonose) [inline web/templates/teacher/form.html.eex]: This status will be removed as well.

File: web/templates/student/index.html.eex [hunk 9]
```diff
@@ -1,39 +1,63 @@
...
-      <td> Email</td>
-      <td> Student ID </td>
-      <td> Participation Type</td>
-      <td> Status </td>
-      <td> Comments </td>
-
+      <th class="mdl-data-table__cell--non-numeric">Name</th>
+      <th class="mdl-data-table__cell--non-numeric">Nickname</th>
+      <th class="mdl-data-table__cell--non-numeric">Email</th>
+      <th class="mdl-data-table__cell--non-numeric">Status</th>
```
Comment (rhnonose) [inline web/templates/student/index.html.eex]: This status will be removed as well.

File: web/templates/student/index.html.eex [hunk 10]
```diff
@@ -1,39 +1,63 @@
...
+          <%= student.name %> <%= student.family_name %>
+        </td>
+        <td class="mdl-data-table__cell--non-numeric">
+          <%= student.nickname %>
+        </td>
+        <td class="mdl-data-table__cell--non-numeric">
+          <%= student.email %>
+        </td>
+        <td class="mdl-data-table__cell--non-numeric">
+          <%= student.status %>
```
Comment (rhnonose) [inline web/templates/student/index.html.eex]: This status will be removed as well.

File: web/templates/student/form.html.eex [hunk 11]
```diff
@@ -1,59 +1,37 @@
...
+                <p>Oops, something went wrong! Please check the errors below.</p>
+              </div>
+            <% end %>
+            <%= CoursePlanner.SharedView.form_text f, :name %>
+            <%= CoursePlanner.SharedView.form_text f, :family_name %>
+            <%= CoursePlanner.SharedView.form_text f, :nickname %>
+            <%= CoursePlanner.SharedView.form_text f, :email, required: true %>
+            <%= CoursePlanner.SharedView.form_text f, :student_id, label: "Student ID" %>
+            <%= CoursePlanner.SharedView.form_select f, :participation_type, participation_types %>
+            <%= CoursePlanner.SharedView.form_select f, :status, statuses %>
```
Comment (rhnonose) [inline web/templates/student/form.html.eex]: This status will be removed as well.

File: web/templates/coordinator/form.html.eex [hunk 12]
```diff
@@ -1,35 +1,33 @@
...
+            <%= if @changeset.action do %>
+              <div class="form-error">
+                <p>Oops, something went wrong! Please check the errors below.</p>
+              </div>
+            <% end %>
+            <%= CoursePlanner.SharedView.form_text f, :name %>
+            <%= CoursePlanner.SharedView.form_text f, :family_name %>
+            <%= CoursePlanner.SharedView.form_text f, :email, required: true %>
+            <%= CoursePlanner.SharedView.form_text f, :phone_number %>
+            <%= CoursePlanner.SharedView.form_select f, :status, statuses %>
```
Comment (rhnonose) [inline web/templates/coordinator/form.html.eex]: This status will be removed as well.

File: web/templates/volunteer/form.html.eex (hunk 6 above)
Comment (laszlokardinal-dina) [inline web/templates/volunteer/form.html.eex]: done

File: web/templates/student/index.html.eex (hunk 9 above)
Comment (laszlokardinal-dina) [inline web/templates/student/index.html.eex]: done

**Comment by rhnonose** [comment]
What about the tests? :)

**Pull request merged**